lib/exchange/provider/trocador_exchange_provider.dart

```diff
@@ -12,11 +12,25 @@ import 'package:cw_core/crypto_currency.dart';
 import 'package:http/http.dart';
 
 class TrocadorExchangeProvider extends ExchangeProvider {
-  TrocadorExchangeProvider({this.useTorOnly = false})
-      : _lastUsedRateId = '',
+  TrocadorExchangeProvider({this.useTorOnly = false, this.providerStates = const {}})
+      : _lastUsedRateId = '', _provider = [],
         super(pairList: supportedPairs(_notSupported));
 
   bool useTorOnly;
+  final Map<String, bool> providerStates;
+
+  static const List<String> availableProviders = [
+    'Swapter',
+    'StealthEx',
+    'Simpleswap',
+    'ChangeNow',
+    'FixedFloat',
+    'LetsExchange',
+    'Exolix',
+    'Godex',
+    'Exch',
+    'CoinCraddle'
+  ];
```

Comment:
do they have an API to get this data dynamically?

Reply:
I didn't find an API for fetching a list of providers. The new_rate method fetches a new list of providers each time for a specific pair of assets and amount.